checker/src/org/checkerframework/checker/index/upperbound/UpperBoundAnnotatedTypeFactory.java

```diff
@@ -737,16 +737,14 @@ private UBQualifier fromLessThanOrEqual(
                 continue;
             }
             CFStore store = getStoreBefore(tree);
-            if (store != null) {
```

Comment:
Why can't the store be null here?

Reply:
When would the Store for `tree` not be available? It looked like this is always done for parts of the code for which we have a Store. I don't think a NPE is raised here, is it?

Why do we need to look up the Store within this loop? Could this be hoisted out of the loop and checked for once?

Reply:
org.checkerframework.framework.type.GenericAnnotatedTypeFactory#getStoreBefore(org.checkerframework.dataflow.cfg.node.Node) might return null, so getStoreBefore(Tree) might return null, too.  But I was mostly wondering if something changed in this pull request that would make this redundant.

Reply:
But the Store is only null if one uses the wrong tree. I rather have a NPE than silently doing nothing.